### PR TITLE
Delimiter and encoding options (plus testing updates)

### PR DESCRIPTION
**Pull request by kjmerf**
This PR is regarding issue #20. 

I added two test files into the demo_data folder:

- titanic_pipe.csv
- titanic_utf16.csv

These can be processed with the respective commands:

```
$ scroll --sep '|' datascroller/datascroller/demo_data/titanic_pipe.csv
$ scroll --encoding utf-16 datascroller/datascroller/demo_data/titanic_utf16.csv
```

In addition, I split the test.sh file into several:

- test.sh, now just used for pytest
- lint.sh, used for the flake8 tests
- format.sh, applies autopep8 to python files in order to pass flake8 tests

### CODE REVIEW

File: datascroller/command_line.py
Comment (kjmerf) [inline datascroller/command_line.py:32]: I imagine this change will be met with some resistance. Splitting the create parser step into its own function makes testing (slightly) easier as we can test the component parts. However, I don't feel strongly about this change if we want to revert.

File: datascroller/scroller.py
Comment (kjmerf) [inline datascroller/scroller.py:55]: Formatting updates according to pep-8 standard. Let's decide if this is too strict.

File: datascroller/scroller.py
Comment (kjmerf) [inline datascroller/scroller.py:368]: The change here is the only substantive change in this file

File: format.sh
Comment (kjmerf) [inline format.sh:1]: Just run ./format.sh to format code in repo. If you run this before ./lint.sh, then the linting tests will pass.

File: lint.sh
Comment (kjmerf) [inline lint.sh:1]: ./lint.sh to perform flake8 test

File: setup.cfg
Comment (kjmerf) [inline setup.cfg:4]: No need to have configuration here as we'll pass these options in the commands

File: test.sh
Comment (kjmerf) [inline test.sh:3]: This file now just runs pytests, of which there are currently three

File: test/test_command_line.py
Comment (kjmerf) [inline test/test_command_line.py:1]: simple tests to make sure we can parse valid arguments. We should probably add negative tests here as well.

**Comment by johncmerfeld** [comment]
Delimiter and encoding functionality works for me!

Weirdly, I still get this screen when I run `scroll --sep '|' datascroller/demo_data/titanic.csv :
image

File: datascroller/scroller.py
Comment (johncmerfeld) [inline datascroller/scroller.py:373]: Would it make sense to wrap this in a try/catch block? Just so we can handle `ValueError` and `UnicodeError` more gracefully if we want?

**Comment by johncmerfeld** [comment]
Testing functionality seems good too. Was able to get it to fail by breaking the demo script. Definitely need to figure out how we can leverage these but glad the framework is there now.

File: datascroller/scroller.py
Comment (kjmerf) [reply datascroller/scroller.py:373]: I might at the parsing step. By the time you get here, we assume the call is clean because, well, the last step is supposed to validate it. If there is a further error internal to pandas I'm not sure I'd change the error message.

**Comment by kjmerf** [comment]
Glad you were able to get it to work. I think it's really powerful but like I said over email, we are still lacking a way to test our core functions, like command_line.run_scroll.py

**Comment by kjmerf** [comment]
Also @johncmerfeld do I need to update version in the settings file?

**Comment by johncmerfeld** [comment]
> Also @johncmerfeld do I need to update version in the settings file?

Probably @baogorek's final call but I'd say yeah? Or I guess it depends. If we haven't released 1.2, we should do a new release after this PR. If we have, we should call it 1.3. I'm not totally sure where we're at.

File: datascroller/scroller.py
Comment (johncmerfeld) [reply datascroller/scroller.py:373]: Makes sense. I guess I'm thinking more of a valid call of the function but where the arguments don't match the file. I guess seeing the full stack trace from pandas isn't the worst thing, but we might prevent a few headaches if we write our own. But you're right, maybe that's counterproductive.

**Comment by baogorek** [comment]
Latest release is 1.1.0 so we could release 1.2 after this PR. Don't feel like you have to wait for me to merge to master. The branch policy is that someone has to do a review, so @johncmerfeld you could create a review, approve, and then @kjmerf can merge. I trust you both.

I'll also show you how to release to PyPi next time we get on hangouts.

File: datascroller/scroller.py
Comment (johncmerfeld) [reply datascroller/scroller.py:55]: I don't have a problem changing existing code to pep-8. Whether we demand new code adhere to it can be discussed.

**Review by johncmerfeld — APPROVED** [review]
Good to merge, I think! Everything worked fine for me. This week let's talk about test cases and how to develop in a test-driven way. We're setting ourselves up really well with this framework. Thanks, Kevin!

**Comment by johncmerfeld** [comment]
Will delete #25

**Pull request merged**
